src/shared/azureagent/roleassignments.go

```diff
@@ -134,13 +112,5 @@ func (a *Agent) FindRoleDefinitionByName(ctx context.Context, scope string, role
 		}
 	}
 
-	missingRoles := lo.Filter(roleNames, func(roleName string, _ int) bool {
-		_, exists := roleDefinitionsByName[roleName]
-		return !exists
-	})
-	if len(missingRoles) > 0 {
-		return nil, errors.Errorf("azure role definitions not found: %s", missingRoles)
-	}
-
```

Comment:
@orishoshan Caused an error when tested, doesn't seem like it's needed with new logic (doesn't exist - move on)